pkg/certificate/manager_test.go

```diff
@@ -68,14 +54,55 @@ var _ = Describe("Test Certificate Manager", func() {
 	})
 })
 
+var _ = Describe("Test Rotor", func() {
+
+	cn := CommonName("foo")
+
+	Context("Testing rotating expiring certificates", func() {
+
+		validityPeriod := -1 * time.Hour // negative time means this cert has already expired -- will be rotated asap
+
+		stop := make(chan struct{})
+		defer close(stop)
+		msgBroker := messaging.NewBroker(stop)
+		certManager, err := NewManager(&Certificate{}, &fakeIssuer{}, validityPeriod, msgBroker)
+		certManager.Start(5*time.Second, stop)
+		Expect(err).ToNot(HaveOccurred())
+
+		certA, err := certManager.IssueCertificate(cn, validityPeriod)
+		certRotateChan := msgBroker.GetCertPubSub().Sub(announcements.CertificateRotated.String())
+
+		It("issued a new certificate", func() {
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		It("rotates certificate", func() {
+			certManager.Start(5*time.Second, stop)
+			done := make(chan interface{})
+
+			start := time.Now()
+			// Wait for one certificate rotation to be announced and terminate
+			<-certRotateChan
+			close(done)
+
+			fmt.Printf("It took %+v to rotate certificate %s\n", time.Since(start), cn)
+			newCert, err := certManager.IssueCertificate(cn, validityPeriod)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(newCert.GetExpiration()).ToNot(Equal(certA.GetExpiration()))
+			Expect(newCert).ToNot(Equal(certA))
+		})
+	})
+
+})
```

Comment:
Since I was playing with this anyway, here's where I ended up with this test:
```go
func TestRotor(t *testing.T) {
	assert := tassert.New(t)

	cn := CommonName("foo")
	validityPeriod := -1 * time.Hour // negative time means this cert has already expired -- will be rotated asap

	stop := make(chan struct{})
	defer close(stop)
	msgBroker := messaging.NewBroker(stop)
	certManager, err := NewManager(&Certificate{}, &fakeIssuer{}, validityPeriod, msgBroker)
	certManager.Start(5*time.Second, stop)
	assert.NoError(err)

	certA, err := certManager.IssueCertificate(cn, validityPeriod)
	assert.NoError(err)
	certRotateChan := msgBroker.GetCertPubSub().Sub(announcements.CertificateRotated.String())

	start := time.Now()
	// Wait for one certificate rotation to be announced and terminate
	<-certRotateChan

	fmt.Printf("It took %+v to rotate certificate %s\n", time.Since(start), cn)
	newCert, err := certManager.IssueCertificate(cn, validityPeriod)
	assert.NoError(err)
	assert.NotEqual(certA.GetExpiration(), newCert.GetExpiration())
	assert.NotEqual(certA, newCert)
}
```